### PR TITLE
feat: cache

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR adds a caching layer per { hostname, recordType } pair. For simplicity, the cache ttl value specified is the minimum of the available records.

### CODE REVIEW

**Comment by vasco-santos** [comment]
cc @jacobheun

File: src/index.js [hunk 1]
```diff
@@ -3,7 +3,13 @@ const debug = require('debug')
 const log = debug('dns-over-http-resolver')
 log.error = debug('dns-over-http-resolver:error')
 
-const { buildResource, fetch } = require('./utils')
+const Cache = require('timed-cache')
```
Comment (jacobheun) [inline src/index.js:6]: This module is an unbounded cache and there's no way to clear it. The DNS max ttl is `2147483647` which is something like 68 years (although encountering this should be extremely rare). The memory footprint should be fairly small and in libp2p we likely won't encounter enough domains to create a problem in between restarts, but we should really bound this (lru based ttl cache) or at least have an easy way of clearing it.

Comment (vasco-santos) [reply src/index.js]: good point! I changed to [receptacle](https://github.com/DylanPiercey/receptacle) as it seems the smallest package available with this requirements and a considerable activity.

**Review by jacobheun — APPROVED** [review]
LGTM. 

Out of scope for this PR, but it might be good to look at doing a shuffle approach in the future when there are multiple servers, to avoid flooding the first server.

**Comment by vasco-santos** [comment]
@jacobheun that would be q good improvement! I will create an issue for tracking it and add it to my todo list, thanks

**Pull request merged**